package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,578 @@
+                       [cp * st, sp * st, ct]])
+        pos2 = Rz.dot(p2 - p1)
+        # Angle about the rotation axis
+        omega = np.arctan2(pos2[1], pos2[0])
+        root_based = np.concatenate((p0, [phi, theta, omega, r01, r12, a012]))
+
+        # Calculate internal coordinates from the torsion list
+        bonds = calc_bonds(self._ag1.positions,
+                           self._ag2.positions,
+                           box=self._ag1.dimensions)
+        angles = calc_angles(self._ag1.positions,
+                             self._ag2.positions,
+                             self._ag3.positions,
+                             box=self._ag1.dimensions)
+        torsions = calc_dihedrals(self._ag1.positions,
+                                  self._ag2.positions,
+                                  self._ag3.positions,
+                                  self._ag4.positions,
+                                  box=self._ag1.dimensions)
+        # When appropriate, calculate improper torsions
+        shift = torsions[self._primary_torsion_indices]
+        shift[self._unique_primary_torsion_indices] = 0.
+        torsions -= shift
+        # Wrap torsions to between -np.pi and np.pi
+        torsions = ((torsions + np.pi) % (2 * np.pi)) - np.pi
+
+        self.bat[self._frame_index,:] = \
+            np.concatenate((root_based, bonds, angles, torsions))
+
+    def load(self, filename, start=None, stop=None, step=None):
+        """Loads the bat trajectory from a file in numpy binary format
+
+        Parameters
+        ----------
+        filename : str
+            name of numpy binary file
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+
+        See Also
+        --------
+        save: Saves the bat trajectory in a file in numpy binary format
+        """
+        logger.info("Choosing frames")
+        self._setup_frames(self._trajectory, start, stop, step)
+
+        logger.info("Loading file")
+        self.bat = np.load(filename)
+
+        # Check array dimensions
+        if self.bat.shape!=(self.n_frames, 3*self._ag.n_atoms):
+          raise ValueError('Dimensions of array in loaded file, ' + \
+              f'({self.bat.shape[0]},{self.bat.shape[1]}), differ from ' + \
+              f'required dimensions of ({self.n_frames, 3*self._ag.n_atoms})')
+        # Check position of initial atom
+        for i, ts in enumerate(self._trajectory[self.start:self.stop:self.step]):
+            self._frame_index = i
+            self._ts = ts
+            self.frames[i] = ts.frame
+            self.times[i] = ts.time
+            if (self.bat[i,:3] != self._root[0].position).any():
+                raise ValueError('Position of initial atom in file ' + \
+                    'inconsistent with current trajectory.')
+        return self
```

Comment:
This is an expensive test: re-analyzing the whole trajectory... I think what you're doing here is the gold standard. If you think it's still reasonable safe to just check
- root atom in frame `start`
- length of trajectory = length of bat
then perhaps the cheaper test is ok? But that's your decision here. Think about what your and the users' use cases are going to be.

Reply:
If you keep the long test, add it to the documentation to tell the user why loading will take a long time. In this case I would also wrap the trajectory in `lib.log.ProgressBar` and add the `verbose` kwarg to mimic what `run()` provides.

Reply:
I'll switch to the short sanity check.